### PR TITLE
changed pdf generator

### PR DESCRIPTION
**Pull request by jahnavisana**

### CODE REVIEW

File: vevestaX/vevesta.py [hunk 1]
```diff
@@ -791,9 +790,18 @@ def __EDAForPandas(self, fileName):
         plt.suptitle('Feature Histogram',fontsize=20)
         plt.savefig(os.path.join(directoryToDumpData, FeatureHistogramImageFile), bbox_inches='tight', dpi=100)
         plt.close()
+        pdffile="EDA.pdf"
+        images=[ValueImageFile,ValueRatioImageFile,NumericalFeatureDistributionImageFile,NonNumericFeaturesImgFile,FeatureHistogramImageFile,OutliersImageFile]
+        a4inpt = (img2pdf.mm_to_pt(210),img2pdf.mm_to_pt(297))
```
Comment (Priyanka-Vevesta) [inline vevestaX/vevesta.py:795]: use better variable name for a4inpt and layout_fun

**Review by Priyanka-Vevesta — CHANGES_REQUESTED** [review]

File: vevestaX/vevesta.py (hunk 1 above)
Comment (jahnavisana) [reply vevestaX/vevesta.py]: changed the variables

File: vevestaX/vevesta.py [hunk 2]
```diff
@@ -46,19 +47,20 @@ def __init__(self, speedUp=False):
         self.__variables = {}
         self.__filename = self.get_filename()
         self.__sampleSize = 0
+        self.__ycolumn=None
```
Comment (Priyanka-Vevesta) [inline vevestaX/vevesta.py:50]: It will be  self.Y

File: vevestaX/vevesta.py [hunk 3]
```diff
@@ -280,16 +282,26 @@ def endModelling(self):
     # create alias of method modellingStart and modellingEnd
     start = startModelling
     end = endModelling
+    @property
+    def Y(self):
```
Comment (Priyanka-Vevesta) [inline vevestaX/vevesta.py:286]: WHy is this a property ?

File: vevestaX/vevesta.py [hunk 4]
```diff
@@ -789,46 +802,72 @@ def __EDAForPandas(self, fileName):
         [x.tick_params(axis='x', labelrotation=90) for x in fig.ravel()]
         plt.plot()
         plt.suptitle('Feature Histogram',fontsize=20)
-        plt.savefig(os.path.join(directoryToDumpData, FeatureHistogramImageFile), bbox_inches='tight', dpi=100)
+        plt.savefig(join(directoryToDumpData, FeatureHistogramImageFile), bbox_inches='tight', dpi=100)
         plt.close()
+        
+        #Probability Density Function
+        numericDataframe = self.__data.select_dtypes(include='number')  
+        if self.__ycolumn!=None:
```
Comment (Priyanka-Vevesta) [inline vevestaX/vevesta.py:810]: if y is regression what happens then i dont see the implementation of what we discussed

File: vevestaX/vevesta.py [hunk 5]
```diff
@@ -789,46 +802,72 @@ def __EDAForPandas(self, fileName):
...
+            k=1
+            fig = plt.figure(figsize=(20,15))
+            for i in numericDataframe:
+                if i!=self.__ycolumn:
+                    ax = fig.add_subplot(4,(len(numericDataframe.columns)//4)+1, k)
+                    frequency=self.__data[self.__ycolumn].value_counts()
+                    frequency=dict(frequency)
+                    frequency=list(frequency.keys())[0:10]
```
Comment (Priyanka-Vevesta) [inline vevestaX/vevesta.py:818]: add a check if the value isnt int or string but float then u shouldnt plot it.

File: vevestaX/vevesta.py [hunk 6]
```diff
@@ -1071,12 +1120,12 @@ def __git_commit(self, git_token, repo_name, branch_name='main', commitMessage=N
...
+            except GithubException:
                 if commitMessage is None:
                     commitMessage = 'added ' + file_name
                 repo.create_file(file_name, commitMessage, file_content, branch=branch_name)
 
     def __find_repo(self, github_user, repo_name):
         all_repos = github_user.get_repos()
         repos = list(filter(lambda r: r.name.casefold() == repo_name.casefold(), all_repos))
-        return repos[0] if len(repos) > 0 else None
+        return repos[0] if len(repos) > 0 else None
```
Comment (Priyanka-Vevesta) [inline vevestaX/vevesta.py:1131]: Add new line in the end.

**Review by Priyanka-Vevesta — CHANGES_REQUESTED** [review]

File: vevestaX/vevesta.py (hunk 5 above)
Comment (Priyanka-Vevesta) [reply vevestaX/vevesta.py:818]: i think you can use this as regression filter

File: vevestaX/vevesta.py (hunk 6 above)
Comment (jahnavisana) [reply vevestaX/vevesta.py]: added maam

File: vevestaX/vevesta.py (hunk 5 above)
Comment (jahnavisana) [reply vevestaX/vevesta.py:818]: yes maam i have added the filter

File: vevestaX/vevesta.py [hunk 7]
```diff
@@ -807,17 +807,17 @@ def __EDAForPandas(self, fileName):
...
             k=1
             fig = plt.figure(figsize=(20,15))
             for i in numericDataframe:
-                if i!=self.__ycolumn:
+                if i!=self.__Y:
                     ax = fig.add_subplot(4,(len(numericDataframe.columns)//4)+1, k)
-                    frequency=self.__data[self.__ycolumn].value_counts()
+                    frequency=self.__data[self.__Y].value_counts()
                     frequency=dict(frequency)
                     frequency=list(frequency.keys())[0:10]
```
Comment (Priyanka-Vevesta) [inline vevestaX/vevesta.py:818]: Have you sorted this to take top 10 frequencies ?

File: vevestaX/vevesta.py [hunk 8]
```diff
@@ -1080,3 +1129,5 @@ def __find_repo(self, github_user, repo_name):
         all_repos = github_user.get_repos()
         repos = list(filter(lambda r: r.name.casefold() == repo_name.casefold(), all_repos))
         return repos[0] if len(repos) > 0 else None
+
+
```
Comment (Priyanka-Vevesta) [inline vevestaX/vevesta.py:1133]: remove extra lines added..only one line extra to be added after the like 1131.

**Review by Priyanka-Vevesta — CHANGES_REQUESTED** [review]

File: vevestaX/vevesta.py (hunk 7 above)
Comment (jahnavisana) [reply vevestaX/vevesta.py:818]: By default, value_counts will sort the data by numeric count in descending order.

**Review by pavanVevesta — APPROVED** [review]

**Review by pavanVevesta — APPROVED** [review]

File: vevestaX/vevesta.py [hunk 9]
```diff
@@ -288,10 +288,18 @@ def Y(self):
...
     @Y.setter
     def Y(self,value):
-        if value in self.__data.columns:
-            self.__Y=value
-        else:
-            print("Column not found")
+        if isinstance(value,Series):
+            if value.size==self.__sampleSize:
+                self.__Y=value
+                self.__v=None
```
Comment (Priyanka-Vevesta) [inline vevestaX/vevesta.py:294]: use __YcolumnName for __v. dont use names no one understands, it crucial step to write good code.

**Review by Priyanka-Vevesta — CHANGES_REQUESTED** [review]

**Review by Priyanka-Vevesta — APPROVED** [review]

**Pull request merged**